GraphResponse.md

```diff
@@ -7,6 +7,248 @@ Provide a decorator for the native response message to enable attaching Graph sp
+
+## The GraphResponse Object
+
+The GraphResponse object can wrap the native HttpResponseMessage object so that customers can use the familiar HttpResponseMessage members. 
+GraphResponse will be part of Microsoft.Graph.Core and it will expose the headers,Content and status code while also keeping reference to the Request object as below.
+
+
+### GraphResponse
+```csharp
+
+    /// <summary>
+    /// The GraphResponse Object
+    /// </summary>
+    public class GraphResponse : IDisposable
+    {
+        /// <summary>
+        /// The GraphResponse Constructor
+        /// </summary>
+        /// <param name="iBaseRequest">The Request made for the response</param>
+        /// <param name="httpResponseMessage">The response</param>
+        public GraphResponse(IBaseRequest iBaseRequest,HttpResponseMessage httpResponseMessage)
+        {
+            this.httpResponseMessage = httpResponseMessage;
+            this.BaseRequest = iBaseRequest;
+        }
+
+        private readonly HttpResponseMessage httpResponseMessage;
+        
+        /// <summary>
+        /// The Response Status code
+        /// </summary>
+        public HttpStatusCode StatusCode => httpResponseMessage.StatusCode;
+        
+        /// <summary>
+        /// The Response Content
+        /// </summary>
+        public HttpContent Content => httpResponseMessage.Content;
+        
+        /// <summary>
+        /// The Response Headers
+        /// </summary>
+        public HttpResponseHeaders HttpHeaders => httpResponseMessage.Headers;
+
+        /// <summary>
+        /// The reference to the Request
+        /// </summary>
+        public IBaseRequest BaseRequest;
+
+        /// <summary>
+        /// Get the native Response Message
+        /// </summary>
+        /// <returns></returns>
+        public HttpResponseMessage ToHttpResponseMessage()
+        {
+            return httpResponseMessage;
+        }
+
+        /// <summary>
+        /// Cleanup
+        /// </summary>
+        public void Dispose()
+        {
+            httpResponseMessage?.Dispose();
+        }
+    }
+
+```
+
+### GraphResponse< T >
+
+This derives from non generic GraphResponse and provides a method of deserialization using a function called `GetResponseObjectAsync()`
+
+```csharp
+
+    /// <summary>
+    /// The GraphResponse Object
+    /// </summary>
+    public class GraphResponse<T> : GraphResponse
+    {
+        /// <summary>
+        /// The GraphResponse Constructor
+        /// </summary>
+        /// <param name="iBaseRequest">The Request made for the response</param>
+        /// <param name="httpResponseMessage">The response</param>
+        public GraphResponse(IBaseRequest iBaseRequest, HttpResponseMessage httpResponseMessage)
+            : base(iBaseRequest, httpResponseMessage)
+        {
+        }
+
+        /// <summary>
+        /// Gets the deserialized object 
+        /// </summary>
+        public async Task<T> GetResponseObjectAsync()
```

Comment:
>GetResponseObjectAsync [](start = 29, length = 22)

should we call this Value to align with how the Azure SDK seems to do it [https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/Response%7BT%7D.cs](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/Response%7BT%7D.cs)

Reply:
I think we should keep this function name as it exposes the action of deserialization to T rather than simple access to the object.